pkg/api/types.go

```diff
@@ -940,10 +940,13 @@ type ContainerStatus struct {
 	Ready bool `json:"ready"`
 	// Note that this is calculated from dead containers.  But those containers are subject to
 	// garbage collection.  This value will get capped at 5 by GC.
-	RestartCount int    `json:"restartCount"`
-	Image        string `json:"image"`
-	ImageID      string `json:"imageID"`
-	ContainerID  string `json:"containerID,omitempty"`
+	RestartCount      int              `json:"restartCount"`
+	Image             string           `json:"image"`
+	ImageID           string           `json:"imageID"`
+	ContainerID       string           `json:"containerID,omitempty"`
+	ImageCreationTime unversioned.Time `json:"imageCreationTime,omitempty"`
```

Comment:
You can copy Image and ImageID into a new struct.  Populate both fields on
output and prefer the old fields if provided on input.

Or you can just leave those 2 and make a struct for new fields.

or Yuju or Dawn might decide it's not worthwhile.  Just an idea.

On Fri, Jan 15, 2016 at 10:30 AM, Nir Dothan <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/19695#discussion_r49886931>
> :
>
> > @@ -940,10 +940,13 @@ type ContainerStatus struct {
> >  	Ready bool `json:"ready"`
> >  	// Note that this is calculated from dead containers.  But those containers are subject to
> >  	// garbage collection.  This value will get capped at 5 by GC.
> > -	RestartCount int    `json:"restartCount"`
> > -	Image        string `json:"image"`
> > -	ImageID      string `json:"imageID"`
> > -	ContainerID  string `json:"containerID,omitempty"`
> > +	RestartCount      int              `json:"restartCount"`
> > +	Image             string           `json:"image"`
> > +	ImageID           string           `json:"imageID"`
> > +	ContainerID       string           `json:"containerID,omitempty"`
> > +	ImageCreationTime unversioned.Time `json:"imageCreationTime,omitempty"`
>
> It makes sense but moving the Image ID to the imageInfo struct would break
> BWC. Would it still make sense if the Image ID was not part of that struct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19695/files#r49886931>.
>